### PR TITLE
Adding example for coords argument in documentation

### LINKED ISSUE #1436: Add an examples of coords argument in documentation
**Describe the bug**
There's this fancy coords argument but not a great explanation on how to use it in the code examples below
https://arviz-devs.github.io/arviz/api/generated/arviz.plot_posterior.html

image

**To Reproduce**
Look at the doc, notice the lack of coords documentation in the code example

**Expected behavior**
Theres an example in the code below

### PR DESCRIPTION
**Pull request by Rishabh261998**
## Description

This PR is created to fix #1436 

## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format

### CODE REVIEW

**Comment by Rishabh261998** [comment]
Hi @OriolAbril could you check this PR and suggest any changes if any. Thank You

File: arviz/plots/posteriorplot.py [hunk 1]
```diff
@@ -149,6 +149,13 @@ def plot_posterior(
         >>> rope = {'mu': [{'rope': (-2, 2)}], 'theta': [{'school': 'Choate', 'rope': (2, 4)}]}
         >>> az.plot_posterior(data, var_names=['mu', 'theta'], rope=rope)
 
+    Adding Coordinates(coords) for distributions selection
+
+    .. plot::
+        :context: close-figs
+
+        >>> coords = {"school": ["Choate","Phillips Exeter"]}
+        >>> az.plot_posterior(data, var_names=["mu", "theta"], coords=coords, rope=(-1, 1))
```
Comment (OriolAbril) [inline arviz/plots/posteriorplot.py:158]: I would remove the rope to focus on `coords` usage.

File: arviz/plots/posteriorplot.py [hunk 2]
```diff
@@ -149,6 +149,13 @@ def plot_posterior(
         >>> rope = {'mu': [{'rope': (-2, 2)}], 'theta': [{'school': 'Choate', 'rope': (2, 4)}]}
         >>> az.plot_posterior(data, var_names=['mu', 'theta'], rope=rope)
 
+    Adding Coordinates(coords) for distributions selection
```
Comment (OriolAbril) [inline arviz/plots/posteriorplot.py:152]: How about ``Using `coords` argument to plot only a subset of data``?

**Comment by OriolAbril** [comment]
Note I have also edited the PR description to have the issue closed automatically when merging, you can read more about this feature here: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by Rishabh261998** [comment]
Did the changes and also added a point in the `CHANGELOG.md`

**Comment by OriolAbril** [comment]
thanks!

**Pull request merged**